File: src/main/java/de/pflugradts/passbird/domain/service/password/provider/PasswordProvider.java
package de.pflugradts.passbird.domain.service.password.provider;

import de.pflugradts.passbird.domain.model.password.PasswordRequirements;
import de.pflugradts.passbird.domain.model.transfer.Bytes;

/**
 * A PasswordProvider generates new Passwords.
 */
public interface PasswordProvider {
    Bytes createNewPassword(PasswordRequirements passwordRequirements);
}

File: src/main/kotlin/de/pflugradts/passbird/domain/service/password/provider/PasswordProvider.kt
package de.pflugradts.passbird.domain.service.password.provider

import de.pflugradts.passbird.domain.model.egg.PasswordRequirements
import de.pflugradts.passbird.domain.model.shell.Shell

interface PasswordProvider {
    fun createNewPassword(passwordRequirements: PasswordRequirements): Shell
}
